design-proposals/status-for-nncp-not-matching-any-node.md

```diff
@@ -0,0 +1,192 @@
+# Status for NNCP not matching any node(s)
+
+## Summary
+
+This proposal is trying to figure out a way on how to handle the status response kubernetes-nmstate is reporting on
+`oc get nncp -oyaml` when the NNCP is not matching any node(s).
+
+## Motivation
+
+Users unfamiliar with the status reports of `oc get nncp -oyaml` might confuse the status report of `status.conditions[].type`
+(Available) and incorrectly assume that a NNCP has been properly applied to node(s) even though it is possibly not
+matching any node(s) `status.conditions[].message` (Policy does not match any node) and `status.conditions[].reason`
+(NoMatchingNode). Tickets like [these](https://issues.redhat.com/browse/OPNET-81) have already come up in the past and
+should therefore be addressed to provide a clear answer.
+
+### User Stories
+
+- As a cluster administrator, I want to be able to easily judge if an NNCP has successfully been rolled out by the status.
+
+### Goals
+
+- Clarify which `status.condition[]` list an applied NNCP not matching any node(s) should have.
+
+### Non-Goals
+
+- Changes to `status.condition[]` on anything else than applied NNCP.
+
+## Proposal
+
+### Solution 1: Add new status [completed/ignored]
+
+Adding a new status will indicate to the user that the NNCP is neither `progressing`, `degraded` nor `available`.
+That status could be called `completed` or `ignored`. `Completed` could indicate that a NNCP has applied but is not "running"
+therefore only `completed` and not `available`. `Ignored` could be used to indicate that the NNCP is valid therefore not
+`degraded` but is being ignored by all nodes
+for any reason (e.g. wrong node selectors).
+
+#### Pros
+
+- It is immediately visible to an admin that the NNCP has been applied successfully but has not affected any node.
+
+#### Cons
+
+- This proposal will add a new status. Scripts from cluster admins checking and relying on the status being (Available) will most likely
+  not work anymore and therefore need some changes.
+
+### Solution 2: Set status degraded
+
+The status could be set to `degraded` to indicate the NNCP is not behaving as expected.
+
+#### Pros
+
+- It is immediately visible to an admin that the applied NNCP might not behave as expected.
+
+#### Cons
+
+- Setting the status to `degraded` could confuse cluster admins that rely on the status indicating an actual problem.
+- It could also cause scripts of cluster admins to behave incorrectly. As they already could rely on the status being `available`
+  even though the NNCP does not match any node(s).
```

Comment:
That is a valid point. One could be devil's advocate and say that if they currently treated NNCPs with NoMatchingNode as good ones (because they are `Available`), it's a chance for them to review those and either fix the selector, or delete NNCPs.